File: api/app/models/hive.py
from enum import Enum as PyEnum
from datetime import datetime
from typing import Optional

from beanie import Document

from .base import TimestampMixin


class HiveStatus(str, PyEnum):
    STRONG = "STRONG"
    ALERT = "ALERT"
    NEEDS_INSPECTION = "NEEDS_INSPECTION"
    WEAK = "WEAK"


class ColonyStrength(str, PyEnum):
    STRONG = "STRONG"
    MODERATE = "MODERATE"
    WEAK = "WEAK"


class QueenStatus(str, PyEnum):
    LAYING = "LAYING"
    NOT_LAYING = "NOT_LAYING"
    MISSING = "MISSING"
    UNKNOWN = "UNKNOWN"


class Temperament(str, PyEnum):
    CALM = "CALM"
    MODERATE = "MODERATE"
    DEFENSIVE = "DEFENSIVE"


class HoneyStores(str, PyEnum):
    FULL = "FULL"
    ADEQUATE = "ADEQUATE"
    LOW = "LOW"
    EMPTY = "EMPTY"


class Hive(Document, TimestampMixin):
    id: str  # type: ignore[assignment]
    name: str
    apiary_id: str
    status: HiveStatus = HiveStatus.STRONG
    last_inspected: datetime
    image_url: Optional[str] = None
    colony_strength: ColonyStrength = ColonyStrength.MODERATE
    queen_status: QueenStatus = QueenStatus.UNKNOWN
    temperament: Temperament = Temperament.MODERATE
    honey_stores: HoneyStores = HoneyStores.ADEQUATE

    class Settings:
        name = "hives"
        indexes = ["apiary_id"]

File: composeApp/src/commonMain/kotlin/com/beekeeper/app/domain/model/Hive.kt
package com.beekeeper.app.domain.model

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class Hive(
    val id: String,
    val name: String,
    val apiaryId: String,
    val status: HiveStatus,
    val lastInspected: LocalDateTime,
    val imageUrl: String?,
    val colonyStrength: ColonyStrength,
    val queenStatus: QueenStatus,
    val temperament: Temperament,
    val honeyStores: HoneyStores
)

enum class HiveStatus {
    STRONG,
    ALERT,
    NEEDS_INSPECTION,
    WEAK
}

enum class ColonyStrength {
    STRONG,
    MODERATE,
    WEAK
}

enum class QueenStatus {
    LAYING,
    NOT_LAYING,
    MISSING,
    UNKNOWN
}

enum class Temperament {
    CALM,
    MODERATE,
    DEFENSIVE
}

enum class HoneyStores {
    FULL,
    ADEQUATE,
    LOW,
    EMPTY
}
